flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/SpillingAdaptiveSpanningRecordDeserializer.java

```diff
@@ -121,7 +134,14 @@ else if (remaining == 0) {
 					}
 				}
 				catch (IndexOutOfBoundsException e) {
-					throw new IOException(BROKEN_SERIALIZATION_ERROR_MESSAGE, e);
+					int bytesRead = this.nonSpanningWrapper.position - oldPosition;
```

Comment:
Actually, now, we cannot ever get to the case that line 133 shows - I'll change that to a `checkState` instead.

I can de-dup a bit but will have to add two position parameters because it is not always a `nonSpanningWrapper` for which we would know what to insert